File: atm.cpp
#include <iostream>

using namespace std;

float determine(int a, float b)
{
    if (a % 5 == 0 && a <= b)
    {
        float total = a + 0.50;
        if (total > b)
        {
            return b;
        }
        else
        {
            return b - total;
        }
    }
    else
    {
        return b;
    }
}


int main(int argc, char **argv)
{
    int amount;
    float bank;
    
    cin >> amount;
    cin >> bank;
    
    float answer = determine(amount, bank);
    cout << answer << endl;
	return 0;
}

File: atm.dart
double solution(int d, double a){
  if(d % 5 == 0 && d <= a){
    double total = d + 0.50;
    if(total > a){
      return a;
    } else {
      return a - total;
    }
  } else 
    return a;
  }
}

void main(){
  int data = 30;
  double amount = 66;
  
  print(solution(data, amount));
}
